test/Auth0.OidcClient.Core.UnitTests/Tokens/NoSignatureVerifier.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Auth0.OidcClient.Tokens;
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Auth0.OidcClient.Core.UnitTests.Tokens
+{
+    internal class NoSignatureVerifier : ISignatureVerifier
+    {
+        private readonly string[] allowedAlgorithms;
+
+        internal NoSignatureVerifier(string[] allowedAlgorithms)
+        {
+            this.allowedAlgorithms = allowedAlgorithms;
+        }
+
+        public JwtSecurityToken VerifySignature(string token)
```

Comment:
is the logic to decode and verify the algorithm name, contained in this particular method, something you might want to move "up" into the interface or maybe an abstract parent class; and re-use it from the other verifier?

Reply:
I don't think so. The `AsymmetricSignatureVerifier` class is a simple `HS256` only class.  Having extra logic there is redundant, takes extra cycles at runtime and could introduce errors/complexity.

None of that matters here in a unit test though and the extra flexibility here is useful.